src/main.rs

```diff
@@ -1,15 +1,47 @@
 use std::fs;
 use std::env;
+use std::process;
+use std::io::prelude::*;
+use std::net::TcpListener;
+use std::net::TcpStream;
 
 fn main() {
     let args: Vec<String> = env::args().collect();
+    if args.len() <= 1 {
+        println!("No .json file was provided as argument!");
+        process::exit(0);
+    }
+
+
+    let json_filename = &args[1];
+    let json_content = fs::read_to_string(json_filename).unwrap();
+
+
+    let ip = "127.0.0.1";
```

Comment:
Looks good 👍, maybe we should change all these let's to const's?
At least for things like ip, port etc. --> also if we do that, name should be all uppercase. (IP, PORT).

Reply:
I was thinking of the possibility the put the ip and port as arrguments later. But yea for the default values we may need constants or maybe put them in a config file. Not sure how to do it rust.

Reply:
Good point, yeah for those things we can have flags later, or maybe a special config file.
For now, I'm still thinking that we should start off with consts etc., maybe DEFAULT_IP and DEFAULT_PORT?

We can also define em outside of main but it's a matter of style/convention, tbd?